app/grandchallenge/algorithms/models.py

```diff
@@ -629,6 +649,118 @@ def create(
 
         return obj
 
+    @property
+    def non_interface_fields(self):
+        return ["algorithm_image", "algorithm_model", "creator", "time_limit"]
```

Comment:
Unsure. It would need to be added in two places, the `JobCreateForm` and the `JobPostSerializer`. So it seemed better to lift this up to the model level so as to only define it once, but I agree that this hides it from future developers. The only way to make sure it gets updated in the future would be to check on the model that the provided data doesn't contain any keys other than interface slugs and what is defined in `non_interface_fields`. Or do you see a different way?

Reply:
I'd say yes as they're properties of two independent forms. It may even be better to auto generate this somehow for each form so that when a new field is added it is updated. Or, at the very least cover it by tests.

Reply:
I will add a test for them for now. Auto-generation would probably take me too much to implement right now.